src/vm/lib/metadata/agent.js

```diff
@@ -749,14 +840,15 @@ MetadataAgent.prototype.createKVMServer = function (zopts, callback) {
         // When the stream closes, we'll delete from zoneConnections so that on
         // next boot (or periodic scan if for some reason we got closed while
         // the zone was actually running) we re-create.
-        zlog.info('stream closed');
+        zlog.info('stream closed on fd %d', fd);
         delete self.zoneConnections[zopts.zone];
     });
 
     kvmstream.connect(zopts.sockpath);
 
-    zlog.info('listening on fd ' + kvmstream._handle.fd);
-    self.zoneConnections[zopts.zone].fd = kvmstream._handle.fd;
+    fd = kvmstream._handle.fd;
+    zlog.info('listening on fd %d', fd);
+    self.zoneConnections[zopts.zone].fd = fd;
```

Comment:
As discussed directly with @misterdjules we do this to make it easier to pull out the handles for debugging. Also in the case that there's an error the handle is sometimes cleared immediately but before we've cleaned up entirely. If we crash in this state we might be able to use the fd we know the handler had to help diagnose what went wrong.